cells/lib/ops/revise.nix

```diff
@@ -0,0 +1,65 @@
+{
+  inputs,
+  cell,
+}: let
+  inherit (inputs) nixpkgs;
+  l = nixpkgs.lib // builtins;
+in
+  /*
+  For use with `revisePackage` and `reviseOCI` to build containers in a mono-repo style
+  environment where the source code is contained in the same repository as the std code,
+  specifically so that one may detect meaningful changes to the image via its tag in the
+  special case where the package's output includes the revision of the source code (e.g. for
+  displaying the version to the user).
+
+  Without special processing, this kind of package would cause the OCI image tag to change
+  on each new revision whether the actual contents of the image changed or not. Combined
+  with `std.incl`, one may have a very strong indicator for when the contents of the image
+  actually includes meaningful changes which avoids flooding the remote registry with superflous
+  copies.
+
+  Args:
+  op: Optional function which takes the package as an argument.
+  pkg: The package you wish to revise.
+  fn: Optional functor with a reference to `pkg` if needed by `op`.
+
+  Returns:
+  The package with a clone of itself in the passthru where the expected revision is set to
+  "not-a-commit" for later use by `reviseOCI`.
+  */
+  op: pkg: fn: let
+    result = op (fn pkg);
+    dummy = "not-a-commit";
+    rev = pkg.src.rev or pkg.src.origSrc.rev or dummy;
+  in
+    if pkg ? sansrev || (rev != dummy && result == pkg)
```

Comment:
`sansrev` looks like a contract that could be documented. If not public, I, personally would benefit from a couple of code comments below.

Reply:
the section after the `||` is to tighten the contract from `revisePackage`. The first contract (existence of sansrev) is universal but the second case is only valid in ussage of revisePackage and so that is made explicit since this condition can only be met if the two functors passed in are both the `id` function.

Reply:
I've made an idempotent refactoring  and I don't understand what this function does:
```nix
  op: pkg: fn: let
    result = op (fn pkg);
    dummy = "not-a-commit";
    rev = pkg.src.rev or pkg.src.origSrc.rev or dummy;

    sansrev = let
      pkg' = op (fn (pkg.sansrev or (pkg.override {rev = dummy;})));
      passthru = pkg'.passthru or {} // {outHash = cell.ops.hashOfPath pkg'.outPath;};
    in
      pkg'.overrideAttrs (_: {inherit passthru;});
    passthru = result.passthru or {} // {inherit sansrev;};
  in
    if pkg ? sansrev || (rev != dummy && result == pkg)
    then result.overrideAttrs (_: {inherit passthru;})
    else result
```

Does it do more than one thing, and if so how many?

Reply:
I gain the impression this function could also be called `42` :smile:

Reply:
It's called like `revise mkOerable packages.operable (operable: { /* ... */ })` The functor is necessary to defer the binding of `operable`